lib/vagrant-libvirt/config.rb

```diff
@@ -538,6 +538,15 @@ def _generate_uri
       end
 
       def finalize!
+        @uri = ENV['VAGRANT_LIBVIRT_URI'] if (@uri == UNSET_VALUE && ENV['VAGRANT_LIBVIRT_URI'])
+        if (@uri && @uri != UNSET_VALUE)
+          uri = URI(@uri)
+          @driver = u.scheme.sub('+ssh','') if @driver == UNSET_VALUE
```

Comment:
Yes, I can't see that there is any easy connection back from the uri to which driver should be used, so I'd leave it to the config in the Vagrant file to set that. Maybe the documentation should state that use of any other libvirt backend with a full `@uri` being provided will still need to set `@driver`, otherwise it will default to 'kvm'.

As @pronix mentions you can also reference environment variables within a Vagrantfile, so if someone wanted to provide support for this in a Vagrantfile they can do so. A little dubious about needing to support use of `ENV['VAGRANT_LIBVIRT_URI']` directly.

However what is still definitely required is to ensure that if a `@uri` is provided that the other components are extracted correctly and set the necessary config options that are used elsewhere in the code base, or that the code using those is switched to extract them from the `@uri` instead. So there is still something to be improved in this area.